test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+const http = require('http');
+const StitchMongoFixture = require('../fixtures/stitch_mongo_fixture');
+
+import { buildClientTestHarness, extractTestFixtureDataPoints } from '../testutil';
+
+describe('With a mock server', () => {
+  const TEST_SERVER_URL = 'http://localhost:7000';
+
+  let handleRequest;
+  const server = http.createServer((req, res) => handleRequest(req, res));
+
+  beforeAll(() => server.listen(7000));
+  afterAll(() => server.close());
+
+  describe('And http service functions', () => {
+    let test = new StitchMongoFixture();
+
+    beforeAll(() => test.setup());
+    afterAll(() => test.teardown());
+
+    const SERVICE_TYPE = 'http';
+    const SERVICE_NAME = 'gd2';
+
+    let th;
+    let service;
+
+    beforeEach(async() => {
+      const { apiKey, groupId, serverUrl } = extractTestFixtureDataPoints(test);
+      th = await buildClientTestHarness(apiKey, groupId, serverUrl);
+
+      const httpService = await th
+        .app()
+        .services()
+        .create({ type: SERVICE_TYPE, name: SERVICE_NAME });
+      await th
+        .app()
+        .services()
+        .service(httpService._id)
+        .rules()
+        .create({
+          name: 'rule1',
+          actions: ['get', 'post', 'put', 'patch', 'delete', 'head'],
+          when: {}
+        });
+      service = th.stitchClient.service(SERVICE_TYPE, SERVICE_NAME);
+    });
+
+    afterEach(async() => th.cleanup());
+
+    describe('Submitting a GET request', () => {
+      const requestHandler = (req, res) => {
+        res.writeHead(200, { 'Content-Type': 'text/plan' });
+        res.end('POW!');
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.get(TEST_SERVER_URL);
```

Comment:
would be good to include some coverage for the response headers too. stitch will include that data in the service response, so we could set headers in the test server and check them in the result here.